### PR TITLE
spatial sampling: option to limit sampling along X-direction to a given range

### PR DESCRIPTION
**Pull request by jtbuch**
Generalized `spatial_sampling.py` to also include sampling along X-direction. Possible options include sampling for Z-direction individually or both directions simultaneously.

### CODE REVIEW

**Comment by slayoo** [comment]
Thank you @jtbuch!
Let's aim for covering the new logic with unit tests as a part this PR

**Comment by jtbuch** [comment]
Ah I think I understand what's going on. Will post another commit on Monday after cleaning up the code a little.

**Comment by slayoo** [comment]
@jtbuch 
We use pre-commit to ensure consistent formatting of source files (whitespaces, etc).
To make the pre-commit hooks update code automatically before each commit (ensuring pre-commit will pass on CI), you can do:
```
pip install pre-commit
cd PySDM
pre-commit install
```

For the test coverage, I suggest to add a new test routine in `tests/unit_tests/initialisation/test_spatial_discretisation.py` which can be analogous to:
https://github.com/open-atmos/PySDM/blob/main/tests/unit_tests/initialisation/test_spatial_discretisation.py#L42-L57
but testing the newly introduced logic

**Comment by jtbuch** [comment]
Sounds good, @slayoo! I will update this within the next week.

**Comment by slayoo** [comment]
Thanks!

**Comment by slayoo** [comment]
@jtbuch rebasing onto (or merging in) the current `main` branch should help with the `examples-setup` failure

**Comment by jtbuch** [comment]
@slayoo Thank you for the suggestion! I am trying to set up multiple branches for different cases (keep an eye out for more issues 👀) and I will rebase before the next commit.

**Comment by jtbuch** [comment]
Hi @slayoo I've added a new function locally that tests for both x- and z-directions based on the existing one in https://github.com/open-atmos/PySDM/blob/main/tests/unit_tests/initialisation/test_spatial_discretisation.py#L42-L57 

Is there a file that calls `test_spatial_discretisation.py` where I need to include the function declaration? Or just pushing a commit with the new unit test should suffice?

**Comment by slayoo** [comment]
Thanks @jtbuch !

> Is there a file that calls test_spatial_discretisation.py where I need to include the function declaration? Or just pushing a commit with the new unit test should suffice?

It is enough to add the file to the repo, see https://docs.pytest.org/en/latest/explanation/goodpractices.html#conventions-for-python-test-discovery

**Comment by jtbuch** [comment]
Ah that link was quite helpful, thanks! I have added the unit test to the repo now after verifying that it works locally.

File: PySDM/initialisation/sampling/spatial_sampling.py [hunk 1]
```diff
@@ -8,23 +8,38 @@
...
 class Pseudorandom:  # pylint: disable=too-few-public-methods
     @staticmethod
-    def sample(*, backend, grid, n_sd, z_part=None):
-        dimension = len(grid)
-        n_elements = dimension * n_sd
+    def sample(*, backend, grid, n_sd, z_part=None, x_part=None):
+        n_dims = len(grid)
+        scale_fac = (
+            []
+        )  # scale factor for each dimension; y = scale_factor * eps + affine_factor
```
Comment (slayoo) [inline PySDM/initialisation/sampling/spatial_sampling.py:15]: @jtbuch, what is `eps`?

Comment (jtbuch) [reply PySDM/initialisation/sampling/spatial_sampling.py:15]: `eps` is just my notation for an array of random uniform samples, which in this case is the `positions` array.

Comment (jtbuch) [reply PySDM/initialisation/sampling/spatial_sampling.py:15]: Hi @slayoo : `eps` is just my notation for an array of random uniform samples, which in this case is the `positions` array.  Sorry, I should have been consistent with my notation.

**Review by slayoo — APPROVED** [review]

**Comment by slayoo** [comment]
Thanks @jtbuch !

**Pull request merged**